src/EditorFeatures/Core.Wpf/InlineRename/UI/SmartRename/SmartRenameViewModel.cs

```diff
@@ -151,43 +159,65 @@ private async Task GetSuggestionsTaskAsync(bool isAutomaticOnInitialization, Can
     {
         if (isAutomaticOnInitialization)
         {
-            // ConfigureAwait(true) to stay on the UI thread;
-            // WPF view is bound to _smartRenameSession properties and so they must be updated on the UI thread.
-            await Task.Delay(_smartRenameSession.AutomaticFetchDelay, cancellationToken).ConfigureAwait(true);
+            await Task.Delay(_smartRenameSession.AutomaticFetchDelay, cancellationToken)
+                .ConfigureAwait(false);
         }
 
         if (cancellationToken.IsCancellationRequested || _isDisposed)
         {
             return;
         }
-        _ = await _smartRenameSession.GetSuggestionsAsync(cancellationToken).ConfigureAwait(true);
-        return;
+
+        if (IsUsingSemanticContext)
+        {
+            var document = this.BaseViewModel.Session.TriggerDocument;
+            var smartRenameContext = ImmutableDictionary<string, string[]>.Empty;
+            var editorRenameService = document.GetRequiredLanguageService<IEditorInlineRenameService>();
+            var renameLocations = await this.BaseViewModel.Session.AllRenameLocationsTask.JoinAsync(cancellationToken)
+                .ConfigureAwait(false);
+            var context = await editorRenameService.GetRenameContextAsync(this.BaseViewModel.Session.RenameInfo, renameLocations, cancellationToken)
+                .ConfigureAwait(false);
+            smartRenameContext = ImmutableDictionary.CreateRange<string, string[]>(
+                context
+                .Select(n => new KeyValuePair<string, string[]>(n.Key, n.Value.ToArray())));
+            _ = await _smartRenameSession.GetSuggestionsAsync(smartRenameContext, cancellationToken)
+                .ConfigureAwait(false);
+        }
+        else
+        {
+            _ = await _smartRenameSession.GetSuggestionsAsync(cancellationToken)
+                .ConfigureAwait(false);
+        }
     }
 
     private void SessionPropertyChanged(object sender, PropertyChangedEventArgs e)
     {
-        _threadingContext.ThrowIfNotOnUIThread();
-        // _smartRenameSession.SuggestedNames is a normal list. We need to convert it to ObservableCollection to bind to UI Element.
-        if (e.PropertyName == nameof(_smartRenameSession.SuggestedNames))
+        _ = _threadingContext.JoinableTaskFactory.RunAsync(async () =>
```

Comment:
this seems ot be kicking off untracked async work.  that will prevent this from being integration testable.

Reply:
why did this switch from just doing the work in this, to kicking off async work to go back to the Ui thread to do the work?